core/src/org/pentaho/platform/engine/core/system/PentahoSystem.java

```diff
@@ -506,6 +513,20 @@ private static void initXMLFactories() {
                 xpathToXMLFactoryNodes ) );
           }
         }
+      } else {
```

Comment:
This is only getting called if PentahoSystem.systemSettingsService is null. I want the java-system-properties loaded all the time. I would move the following code to another method "setSystemProperties(), called right before initXmlFactories() in PentahoSystem.init().